osu.Game/Screens/Play/HUD/GameplayRankDisplay.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Localisation;
+using osu.Game.Configuration;
+using osu.Game.Localisation.HUD;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    public abstract partial class GameplayRankDisplay : CompositeDrawable, IHasCurrentValue<ScoreRank>
```

Comment:
hesitant about this existing, the duplication was inoffensive before and the logic was straightforward, with this abstract class existing people may get dumb ideas

Reply:
Can you elaborate further? What dumb ideas would people have from sharing code between default/legacy pieces in an abstract component?

Reply:
i'm generally bearish on abstract classes these days and prefer duplication if it's duplication of trivial code. later monstrosities like `ScoreTable` get created that cannot be set right without a day wasted on righting it

Reply:
If it was in the other situation like there are only 10 lines of code that were duplicated, I would prefer to just not extract them, but if there are huge amounts of business logic duplications, then I think should probably extract them to a separate class, or function, not only for the better readability of code but also for the better maintenance for the further changes.